lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -67,7 +67,7 @@ function call_manopt_optimizer(
         x0;
         stopping_criterion::Union{Manopt.StoppingCriterion, Manopt.StoppingCriterionSet},
         evaluation::AbstractEvaluationType = Manopt.AllocatingEvaluation(),
-        stepsize::Stepsize = ArmijoLinesearch(M),
+        stepsize::Stepsize = ArmijoLinesearchStepsize(M),
```

Comment:
This can for example stay as before, as I just wrote in the general comment `ArmijoLiineSearch` now introduces a factory (ok it is not a `Stepsise` type then, and when necessary one can `produce_type` on that, see the pattern in action at 

https://github.com/JuliaManifolds/Manopt.jl/blob/013feadf95736639cdedb1bff2c4f1723f7dfe64/src/solvers/gradient_descent.jl#L208-L210

(note the new union there)

as well as

https://github.com/JuliaManifolds/Manopt.jl/blob/013feadf95736639cdedb1bff2c4f1723f7dfe64/src/solvers/gradient_descent.jl#L233C1-L233C45

which “turns” a factory into the concrete thing (here the stepwise) and is the identity if one already provides a stepwise.